timeseries/src/autogluon/timeseries/models/statsmodels/abstract_statsmodels.py

```diff
@@ -0,0 +1,205 @@
+import logging
+import re
+from dataclasses import dataclass
+from functools import partial
+from multiprocessing import cpu_count
+from typing import Any, Dict, List, Optional, Tuple
+
+import pandas as pd
+from joblib import Parallel, delayed
+from statsmodels.tsa.base.tsa_model import TimeSeriesModelResults as StatsmodelsTSModelResults
+
+from autogluon.timeseries.dataset.ts_dataframe import ITEMID, TIMESTAMP, TimeSeriesDataFrame
+from autogluon.timeseries.models.abstract import AbstractTimeSeriesModel
+from autogluon.timeseries.utils.hashing import hash_ts_dataframe_items
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class FittedLocalModel:
+    """Stores the parameters of a fitted Statsmodels time series model.
+
+    FittedLocalModel doesn't store intermediate results and therefore takes up very little disk space when pickled. In
+    contrast, StatsmodelsFittedModel stores the intermediate results and therefore can often take several gigabytes
+    when pickled.
+
+    FittedLocalModel is consumed by AbstractStatsmodelsModel._predict_using_fit_summary to generate predictions.
+    """
+
+    model_name: str
+    model_init_args: Dict[str, Any]
+    parameters: Dict[str, Any]
+
+
+class AbstractStatsmodelsModel(AbstractTimeSeriesModel):
+    """Abstract class for local models that are fitted for each time series in a dataset.
+
+    Parameters
+    ----------
+    path: str
+        directory to store model artifacts.
+    freq: str
+        string representation (compatible with GluonTS frequency strings) for the data provided.
+        For example, "1D" for daily data, "1H" for hourly data, etc.
+    prediction_length: int
+        Number of time steps ahead (length of the forecast horizon) the model will be optimized
+        to predict. At inference time, this will be the number of time steps the model will
+        predict.
+    name: str
+        Name of the model. Also, name of subdirectory inside path where model will be saved.
+    eval_metric: str
+        objective function the model will be scored on, will use mean_wQuantileLoss by default.
+    hyperparameters:
+        various hyperparameters that will be used by model (can be search spaces instead of
+        fixed values). See *Other Parameters* in each inheriting model's documentation for
+        possible values.
+    """
+
+    statsmodels_allowed_init_args: List[str] = []
+    statsmodels_allowed_fit_args: List[str] = []
+    quantile_method_name: str = "pred_int"
+
+    def __init__(
+        self,
+        freq: Optional[str] = None,
```

Comment:
Can we expect that `freq` is always not `None`? We need `freq` when making predictions and computing the `start` and `end` dates for the forecast.

Reply:
the hope is that we always have freq available in the context through `TimeSeriesDataFrame.freq`